Userland/Libraries/LibJS/Runtime/SharedArrayBufferPrototype.cpp

```diff
@@ -0,0 +1,129 @@
+
+    // 4. Let length be O.[[ArrayBufferByteLength]].
+    // 5. Return 𝔽(length).
+    return Value(array_buffer_object->byte_length());
+}
+
+// 25.2.4.3 SharedArrayBuffer.prototype.slice ( start, end ), https://tc39.es/ecma262/#sec-sharedarraybuffer.prototype.toString
+JS_DEFINE_NATIVE_FUNCTION(SharedArrayBufferPrototype::slice)
+{
+    auto& realm = *vm.current_realm();
+
+    // 1. Let O be the this value.
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
+    auto array_buffer_object = TRY(typed_this_value(vm));
+
+    // 3. If IsSharedArrayBuffer(O) is false, throw a TypeError exception.
+    // FIXME: Check for shared buffer
+
+    // 4. Let len be O.[[ArrayBufferByteLength]].
+    auto length = array_buffer_object->byte_length();
+
+    // 5. Let relativeStart be ? ToIntegerOrInfinity(start).
+    auto relative_start = TRY(vm.argument(0).to_integer_or_infinity(vm));
+
+    double first;
+    // 6. If relativeStart is -∞, let first be 0.
+    if (Value(relative_start).is_negative_infinity())
+        first = 0;
+    // 7. Else if relativeStart < 0, let first be max(len + relativeStart, 0).
+    else if (relative_start < 0)
+        first = max(length + relative_start, 0.0);
+    // 8. Else, let first be min(relativeStart, len).
+    else
+        first = min(relative_start, (double)length);
+
+    // 9. If end is undefined, let relativeEnd be len; else let relativeEnd be ? ToIntegerOrInfinity(end).
+    auto relative_end = vm.argument(1).is_undefined() ? length : TRY(vm.argument(1).to_integer_or_infinity(vm));
+
+    double final;
+    // 10. If relativeEnd is -∞, let final be 0.
+    if (Value(relative_end).is_negative_infinity())
+        final = 0;
+    // 11. Else if relativeEnd < 0, let final be max(len + relativeEnd, 0).
+    else if (relative_end < 0)
+        final = max(length + relative_end, 0.0);
+    // 12. Else, let final be min(relativeEnd, len).
+    else
+        final = min(relative_end, (double)length);
+
+    // 13. Let newLen be max(final - first, 0).
+    auto new_length = max(final - first, 0.0);
+
+    // 14. Let ctor be ? SpeciesConstructor(O, %ArrayBuffer%).
+    auto* constructor = TRY(species_constructor(vm, array_buffer_object, realm.intrinsics().shared_array_buffer_constructor()));
+
+    // 15. Let new be ? Construct(ctor, « 𝔽(newLen) »).
+    auto new_array_buffer = TRY(construct(vm, *constructor, Value(new_length)));
+
+    // 16. Perform ? RequireInternalSlot(new, [[ArrayBufferData]]).
+    if (!is<ArrayBuffer>(new_array_buffer.ptr()))
+        return vm.throw_completion<TypeError>(ErrorType::SpeciesConstructorDidNotCreate, "an SharedArrayBuffer");
+    auto* new_array_buffer_object = static_cast<ArrayBuffer*>(new_array_buffer.ptr());
+
+    // 17. If IsSharedArrayBuffer(new) is true, throw a TypeError exception.
+    // FIXME: Check for shared buffer
+
+    // FIXME: 18. If new.[[ArrayBufferData]] is O.[[ArrayBufferData]], throw a TypeError exception.
```

Comment:
This is a simple pointer compare, no reason to not implement it :)

Reply:
Thanks! Confirmed it fixes that issue :^)